src/internal/components/Draggable.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useEffect, useState } from 'react';
+import { zIndex } from '../../styles/constants';
+import { cn } from '../../styles/theme';
+
+type DraggableProps = {
+  children: React.ReactNode;
+  gridSize?: number;
+  startingPosition?: { x: number; y: number };
+  enableSnapToGrid?: boolean;
+};
+
+export default function Draggable({
+  children,
+  gridSize = 1,
+  startingPosition = { x: 20, y: 20 },
+  enableSnapToGrid = true,
```

Comment:
nit: since this is default true, may be an easier developer implementation to add disableSnapToGrid to turn it off vs. enableSnapToGrid={false}

Reply:
ah, sorry i meant to default this to false.

Reply:
in that case you could probably just call it `snapToGrid`